src/InkRouter/Models/PrintAsset.php

```diff
@@ -0,0 +1,195 @@
+ */
+
+/**
+ * @author Vladimir Prudilin <vladimir.prudilin@opensoftdev.ru>
+ */
+class InkRouter_Models_PrintAsset
+{
+    /**
+     * @var integer
+     */
+    private $positionX;
+
+    /**
+     * @var integer
+     */
+    private $positionY;
+
+    /**
+     * @var integer
+     */
+    private $width;
+
+    /**
+     * @var integer
+     */
+    private $height;
+
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var integer
+     */
+    private $rotation;
+
+    /**
+     * @return integer
+     */
+    public function getPositionX()
+    {
+        return $this->positionX;
+    }
+
+    /**
+     * @param integer $positionX
+     * @return InkRouter_Models_PrintAsset
+     */
+    public function setPositionX($positionX)
+    {
+        $this->positionX = $positionX;
+
+        return $this;
+    }
+
+    /**
+     * @return integer
+     */
+    public function getPositionY()
+    {
+        return $this->positionY;
+    }
+
+    /**
+     * @param integer $positionY
+     * @return InkRouter_Models_PrintAsset
+     */
+    public function setPositionY($positionY)
+    {
+        $this->positionY = $positionY;
+
+        return $this;
+    }
+
+    /**
+     * @return integer
+     */
+    public function getWidth()
+    {
+        return $this->width;
+    }
+
+    /**
+     * @param integer $width
+     * @return InkRouter_Models_PrintAsset
+     */
+    public function setWidth($width)
+    {
+        $this->width = (int) $width;
+
+        return $this;
+    }
+
+    /**
+     * @return integer
+     */
+    public function getHeight()
+    {
+        return $this->height;
+    }
+
+    /**
+     * @param integer $height
+     * @return InkRouter_Models_PrintAsset
+     */
+    public function setHeight($height)
+    {
+        $this->height = (int) $height;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getType()
+    {
+        return $this->type;
+    }
+
+    /**
+     * @param string $type
+     * @return InkRouter_Models_PrintAsset
+     */
+    public function setType($type)
+    {
+        $this->type = $type;
+
+        return $this;
+    }
+
+    /**
+     * @return integer
+     */
+    public function getRotation()
+    {
+        return $this->rotation;
+    }
+
+    /**
+     * @param integer $rotation
+     * @return InkRouter_Models_PrintAsset
+     */
+    public function setRotation($rotation)
+    {
+        $this->rotation = (int) $rotation;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function pack()
+    {
+        $writer = new XMLWriter();
+        $writer->openMemory();
+
+        $writer->startElement('print_asset');
+
+        if (isset($this->positionX)) {
+            $writer->writeElement('position_x', $this->positionX);
+        }
+
+        if (isset($this->positionY)) {
+            $writer->writeElement('position_y', $this->positionY);
```

Comment:
also i talked with Profit and it seems that they won't be able to implement support of different origin points in time (if it is expected to be done along with introducing new property)